### PR TITLE
Transformations: Fix `extractFields` throwing Error if one value is undefined or null

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

If one value of a DataFrame field is either `null` or `undefined`, the "Extract Fields" transformation will throw an error, because it can not call `Object.entries(obj)` in https://github.com/grafana/grafana/blob/main/public/app/features/transformers/extractFields/extractFields.ts#L80 with `obj` being `null` or `undefined`. 
This PR simply skips those DataFrame rows.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

File: public/app/features/transformers/extractFields/extractFields.ts [hunk 1]
```diff
@@ -77,6 +77,10 @@ function addExtractedFields(frame: DataFrame, options: ExtractFieldsOptions): Da
       }
     }
 
+    if (!obj) {
```
Comment (leeoniya) [inline public/app/features/transformers/extractFields/extractFields.ts:80]: `obj != null` should be a good deal faster

File: public/app/features/transformers/extractFields/extractFields.ts [hunk 2]
```diff
@@ -77,6 +77,10 @@ function addExtractedFields(frame: DataFrame, options: ExtractFieldsOptions): Da
       }
     }
 
+    if (obj == null) {
```
Comment (matyax) [inline public/app/features/transformers/extractFields/extractFields.ts:80]: I would move this up, at least around line 64.

File: devenv/docker/blocks/loki/data/data.js [hunk 3]
```diff
@@ -195,8 +196,31 @@ async function sendNewLogs() {
 }
 
 async function main() {
-  // we generate both old-logs and new-logs at the same time
-  await Promise.all([sendOldLogs(), sendNewLogs()])
+  function myFunction() {
```
Comment (matyax) [inline devenv/docker/blocks/loki/data/data.js:199]: Is this test code? Why is it called `myFunction`?

Comment (svennergr) [reply devenv/docker/blocks/loki/data/data.js:199]: That happens when you blindly do a `git commit -a -m`... 🙈

File: public/app/features/transformers/extractFields/extractFields.ts (hunk 2 above)
Comment (svennergr) [reply public/app/features/transformers/extractFields/extractFields.ts:80]: Well, `obj` can be changed in line 76, so this needs to be done here.

Comment (matyax) [reply public/app/features/transformers/extractFields/extractFields.ts:80]: Yep, but `const newObj: { [k: string]: unknown } = {};`. Not a big deal in any case 👍

Comment (svennergr) [reply public/app/features/transformers/extractFields/extractFields.ts]: Good spot, Matias! Changed in d2d5dd588ddf93fa0e3bb460a99c194d08acb309

**Pull request merged**